xla/service/gpu/gpu_latency_hiding_scheduler.cc

```diff
@@ -126,19 +126,43 @@ bool IsAsyncPair(const HloInstruction& from, const HloInstruction& target) {
   return IsGpuAsyncStart(from) && IsGpuAsyncDone(target);
 }
 
+// Util function for getting replica groups from different data structures.
+static std::vector<std::vector<int64_t>> get_replica_groups(
```

Comment:
I think there is something like this in the collective ops utils. Can you mve it there and test it separately?